core/src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala

```diff
@@ -482,6 +483,28 @@ object ConditionalScriptSignature
     fromAsm(nestedScriptSig.asm.:+(conditionAsm))
   }
 
+  @scala.annotation.tailrec
+  def apply(
+      nestedScriptSig: ScriptSignature,
+      conditionalPath: ConditionalPath): ConditionalScriptSignature = {
+    conditionalPath match {
+      case ConditionalPath.NoConditionsLeft =>
+        throw new IllegalArgumentException("ConditionalPath cannot be empty")
+      case ConditionalPath.nonNestedTrue =>
```

Comment:
This is really unfortunate that `.nonNestedFalse` and `nonNestedTrue` aren't real ADTs :-(, especially since they are the terminating cases's for how `ConditionalPath` is defined.

Reply:
I would argue that the only terminating case for `ConditionalPath` is `NoConditionsLeft` (as it should be) and that these two vals are just short-hand for exactly what I'm going for in this match case (which is to extract further than just terminal cases)

Reply:
This layout breaks pattern matching exhaustiveness though. Why not just make them `final case object` if they are important?

Reply:
It doesn't break exhaustiveness: I removed the case beneath this and got
```
[warn] /home/nkohen/Desktop/SuredBits/bitcoin-s-core/core/src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala:486:5: match may not be exhaustive.
[warn] It would fail on the following input: ConditionTrue(_)
[warn]     conditionalPath match {
[warn]     ^
[warn] one warning found
```

Essentially when representing branches of a binary tree a single step/leg is going to be more frequent than other things (especially in this context) but it is not trivial or fundamental, only leaves are trivial.